examples/src/main/scala/example/ziohttp/AuthExampleApp.scala

```diff
@@ -7,39 +7,48 @@ import caliban.interop.tapir.{ StreamTransformer, WebSocketHooks }
 import caliban.schema.GenericSchema
 import example.ExampleData._
 import example.{ ExampleApi, ExampleService }
-import io.netty.handler.codec.http.{HttpHeaderNames, HttpHeaderValues}
+import io.netty.handler.codec.http.{ HttpHeaderNames, HttpHeaderValues }
 import zhttp.http._
 import zhttp.service.Server
 import zio._
 import zio.clock._
 import zio.duration._
 import zio.stream._
 
+case object Unauthorized extends RuntimeException("Unauthorized")
+
 trait Auth {
-  def currentUser: ZIO[Any, Throwable, String]
-  def setUser(name: String): ZIO[Any, Throwable, Any]
+  type Unauthorized = Unauthorized.type
+
+  def currentUser: IO[Unauthorized, String]
+  def setUser(name: String): UIO[Unit]
 }
 
 object Auth {
-  val http = FiberRef
-    .make("unknown")
-    .map { ref =>
-      new Auth {
-        def currentUser: ZIO[Any, Throwable, String]        = ref.get
-        def setUser(name: String): ZIO[Any, Throwable, Any] = ref.set(name)
+
+  val http: ULayer[Has[Auth]] =
+    FiberRef
+      .make[String](null)
+      .map { ref =>
+        new Auth {
+          def currentUser: IO[Unauthorized, String] =
+            ref.get.flatMap(v => if (v eq null) ZIO.fail(Unauthorized) else ZIO.succeed(v))
+          def setUser(name: String): UIO[Unit]      = ref.set(name)
+        }
       }
-    }
-    .toLayer
+      .toLayer
 
   object WebSockets {
-    val wsSession = Http.fromEffect(Ref.make[String]("unknown"))
+    private val wsSession = Http.fromEffect(Ref.make[String](null))
```

Comment:
How about `Option[String]` and no null 😄

Reply:
`Option` creates a useless allocation. In terms of performances, `null` is better solution here

Reply:
It's insignificant compared to everything that happens when handling a query. I think it's better that examples look cleaner.

Reply:
Changed for Option